tests/extra/cold.ts

```diff
@@ -0,0 +1,58 @@
+/// <reference path="../../typings/globals/mocha/index.d.ts" />
+/// <reference path="../../typings/globals/node/index.d.ts" />
+import xs from '../../src/index';
+import cold from '../../src/extra/cold';
+import * as assert from 'assert';
+
+describe('cold (extra)', () => {
+  it('should push values to a listener', (done) => {
+    const stream = xs.periodic(100).take(3).compose(cold);
+
+    const expected = [0, 1, 2];
+    const listener = {
+      next (x: number) {
+        assert(x === expected.shift());
+      },
+      error: done,
+      complete () {
+        assert(expected.length === 0);
+        stream.removeListener(listener);
+        done();
+      }
+    };
+
+    stream.addListener(listener);
+  });
+
+  it('should restart the producer for every new listener', (done) => {
+    const stream = xs.periodic(100).take(3).compose(cold);
```

Comment:
Here's why this test fails.
xs.periodic(100) is hot
xs.periodic(100).take(3) is hot
xs.periodic(100).take(3).cold is "cold" but behaves like hot
xs.of(0, 1, 2) is hot
xs.of(0, 1, 2).compose(cold) is **truly cold**
xs.of(0, 1, 2).compose(cold).take(3) is hot

Which means if we want cold behavior to propagate, we need to implement the `ctor()` technique in all core operators.